### PR TITLE
Compose TiltSeries streaming

### PR DESCRIPTION
**Pull request by albertmena**
Behavior:

When a tilt Serie is acquired and there are all the micrographs in the input set, a stack is generated and added the tilt serie to the set of tilt series in the output.

Input: 

- set of micrographs (from alignment protocol...)
- folder with mdoc files with all the tilts

Times out:

- Time for the next tilt (if the tiltserie acquisition has not ended and there is a mdoc file with information of the tilts acquired). 
- Time next micrograph: Time to wait until the next micrograph is processed by the previous protocol (alignment)
- Time next TiltSerie: Time to wait until the next TiltSerie starts to be acquired (time until the mdoc file is created)

### CODE REVIEW

**Comment by azazellochg** [comment]
Hi, I didnt get where the micrographs are coming from? motion correction protocols produce outputTiltSeries. Or you are referring to acquisition in serialem which generates (usually binned) TS mrc stacks? But then you will have 1 mdoc file for each stack.

**Comment by pconesa** [comment]
We are feeding this protocol with SPA motion correction micrographs.

**Comment by azazellochg** [comment]
O_o

Ehm, okay.

**Comment by azazellochg** [comment]
I would then add a point that serialem can rearrange binned TS stacks and mdocs after the tilt-series is collected (https://github.com/scipion-em/scipion-em-tomo/issues/232). I hope this will be supported here as well.

**Comment by pconesa** [comment]
Are talking about 2 options here:

1.- Serial EM does nothing, so MC is done in "Scipion" and TS compose is done on a "non rearranged mdoc"
2.- Serial EM does the rearrange, and I guess the MC too and the stack? So we just register the whole Tilt series?

**Comment by azazellochg** [comment]
serialem does not re-arrange movies in any way since they are individual files, so mc is not affected. I suppose the composition will depend on how you get the info from the mdoc. Are you comparing micname from scipion with movie file name in mdoc?

File: tomo/protocols/protocol_compose_TS.py [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+        """
+        :return: return a sorted by date list of all mdoc files in the path
+        """
+        """ return a sorted by date list of all mdoc files in the path """
+        fpath = self.filesPath.get()
+        self.MDOC_DATA_SOURCE = glob(os.path.join(fpath, '*.mdoc'))
+        self.MDOC_DATA_SOURCE.sort(key=os.path.getmtime)
+        return self.MDOC_DATA_SOURCE
+
+    def readMdoc(self, list_remains):
```
Comment (pconesa) [inline tomo/protocols/protocol_compose_TS.py:171]: I think it will be better to deal with a single mdoc in this step. For resuming options is better.

**Review by pconesa — CHANGES_REQUESTED** [review]
Almost there. Please, could you provide comments at least on the methods that are more complex. I mean doc strings.

And a test.

**Comment by pconesa** [comment]
We do the matching by name in mdoc with the "movie/mic name". So I guess we are covered if is just the mdoc what is rearranged.

**Comment by azazellochg** [comment]
will this support pacetomo (https://github.com/scipion-em/scipion-em-tomo/issues/236)?

**Comment by pconesa** [comment]
Does pacetomo generate mdoc files like serial EM?

**Comment by albertmena** [comment]
The test is pending to do

**Review by pconesa — APPROVED** [review]

**Pull request merged**